### DISCUSSION TITLE
Include model on through association

### DISCUSSION THREAD

**Discussion by hugoobauer** [sequelize/sequelize#15132] [Help & Questions]
I'm creating a new issue from https://github.com/sequelize/sequelize/issues/10828 because it's closed but can't find any solution on the documentation.

I have two models, `User` and `Account` that are linked through a reference table `UserAccountMap`. The linking/through table also contains references to the `Role` table.
```
User <--> UserAccountMap <--> Account
                |
                v
               Role
```
How can I include `Role` table with this kind of query.
```ts
const user = await User.findByPk(userId, {
  include: [{ model: Account }],
});
```
I can't pass `include` to the `through` params.

I tried to add a defaultScope to my `UserAccountMap` through table but it's not including the Role model with the above query ...
UserAccountMap init options :
```ts
defaultScope: {
  include: [{ model: Role }],
},
```

**Comment by ephys** [comment]
You can't do that directly, you need to include the through model (`UserAccountMap`) then include both `Account` and `Role`:

```typescript
const user = await User.findByPk(userId, {
  include: [{
    association: User.assocations.userAccountMaps,
    include: [
      UserAccountMap.associations.accounts,
      UserAccountMap.associations.roles,
    ],
  }],
});
```

This requires defining the associations between `User` and `UserAccountMap`, `Account` and `UserAccountMap`, and `Role` and `UserAccountMap`.

It's what our docs call the "Super Many-to-Many" https://sequelize.org/docs/v6/advanced-association-concepts/advanced-many-to-many/#the-best-of-both-worlds-the-super-many-to-many-relationship (it's done by default in v7)
